cmd/cstor-volume-grpc/app/command/commands.go

```diff
@@ -0,0 +1,280 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-volume-grpc/api"
+	"github.com/openebs/maya/pkg/util"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+var (
+	cmdName = "cstor-volume-grpc"
+	usage   = fmt.Sprintf("%s", cmdName)
+)
+
+// CmdSnaphotOptions holds the options for snapshot
+// create command
+type CmdSnaphotOptions struct {
+	volName  string
+	snapName string
+}
+
+// Validate validates the flag values
+func (c *CmdSnaphotOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please specify an unique name")
+	}
+	if c.snapName == "" {
+		return errors.New("--snapname is missing. Please specify an unique name")
+	}
+
+	return nil
+}
+
+//CreateSnapshot creates snapshots
+func CreateSnapshot(volName, snapName string) (*api.VolumeCommand, error) {
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial(fmt.Sprintf(":%d", api.VolumeGrpcListenPort), grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("did not connect: %s", err)
+	}
+	defer conn.Close()
+
+	c := api.NewRunCommandClient(conn)
+	response, err := c.RunVolumeCommand(context.Background(),
+		&api.VolumeCommand{
+			Command:  api.CmdSnapCreate,
+			Volume:   volName,
+			Snapname: snapName,
+			Status:   "requesting",
+		})
+
+	if err != nil {
+		log.Fatalf("Error when calling RunVolumeCommand: %s", err)
+	}
+	log.Printf("Response from server: %s, %s, %s, %s",
+		response.Command, response.Volume, response.Snapname, response.Status)
+	return response, err
+}
+
+//DestroySnapshot destroys snapshots
+func DestroySnapshot(volName, snapName string) (*api.VolumeCommand, error) {
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial(fmt.Sprintf(":%d", api.VolumeGrpcListenPort), grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("did not connect: %s", err)
+	}
+	defer conn.Close()
+
+	c := api.NewRunCommandClient(conn)
+	response, err := c.RunVolumeCommand(context.Background(),
+		&api.VolumeCommand{
+			Command:  api.CmdSnapDestroy,
+			Volume:   volName,
+			Snapname: snapName,
+			Status:   "requesting",
+		})
+
+	if err != nil {
+		log.Fatalf("Error when calling RunVolumeCommand: %s", err)
+	}
+	log.Printf("Response from server: %s, %s, %s, %s",
+		response.Command, response.Volume, response.Snapname, response.Status)
+	return response, err
+}
+
+// RunSnapshotCreate does tasks related to grpc snapshot create.
+func (c *CmdSnaphotOptions) RunSnapshotCreate(cmd *cobra.Command) error {
+	fmt.Println("Executing volume snapshot create...")
+	resp, err := CreateSnapshot(c.volName, c.snapName)
+	if err != nil {
+		return fmt.Errorf("Snapshot create failed: %v", err)
+	}
+	if resp != nil && strings.Contains(resp.Status, "ERR") {
```

Comment:
So if err == nill, we can still have an error? that seems to be an anti-pattern.

Reply:
have removed this code. now, generating the error from the createsnapshot and destroysnapshot methods upon receiving error response from istgt.